Support collecting logs of log4j, log4j2, and logback in the tracing context (#5914)
apache/skywalking: docs/en/setup/service-agent/java-agent/agent-optional-plugins/Logger-plugin.md

> # Overview

Reviewer: I think we miss a part about how to disable one log framework, such as opening the bridge for log4j only.

Author: I have removed `logconfig.properties` and add some use cases in the document.

`logger-plugin` can store the logs generated by the program during the call, such as the content of the error log, into the span log.  Through the configuration file, you can control the log source (log4j2, logback, log4j), package name, and level.

# Configuration file

> By default, the configuration file is in `apache-skywalking-apm-bin/agent/config//logger-plugin/logconfig.properties`.Of course, **If the file does not exist, the configuration will still be configured with the following default values:**

Reviewer: According to this, we should not have the default `logconfig.properties` in the distribution.

Author: If the configuration file not exists, what should we do? Throws an Exception?

Author: I have removed `logconfig.properties` and add some use cases in the document.

```properties
log4j.packages=*
log4j.level=error
log4j2.packages=*
log4j2.level=error
logback.packages=*
logback.level=error
```

The meaning of the above configuration is as follows:

> 1. It will process information from all logging systems, including `log4j`, `log4j2`, `logback`.

Reviewer: ```suggestion
1. SkyWalking opened the adaptor(bridge) between tracing kernel and log frameworks, including `log4j`, `log4j2`, `logback`.
```

> 2. Only the log information of the `error` level will be processed, and the log information lower than the `error` level will be ignored, including `trace`, `debug`, `info`, `warn`.

Reviewer: ```suggestion
2. Only collect logs at the `error` level, others would be ignored, including `trace`, `debug`, `info`, `warn`.
```

> 3. It will match all packages.

Reviewer: ```suggestion
3. Wouldn't filter the logs by the package name.
```

> # Field description

Reviewer: ```suggestion
# Property description
```

## packages

**package attribute**: Specify the package name of the log that needs log conversion.  **The default value is `*` , and it will match all packages.**

packages value:

* the name of package, eg: `org.apache.skywalking`
* `*`:match all 

**Notice:**

> When matching multiple packages, the names of different packages should be separated by commas.

Reviewer: ```suggestion
When matching multiple packages, the names of different packages should be separated by the comma.
```

## level

**level attribute** : The level of the log for conversion. **By default it is `error` level**.

> The specific sequence of log levels from low to high is as follows:

Reviewer: ```suggestion
The hierarchy order of log levels from low to high is as follows:
```

# Overview

> `logger-plugin` can store the logs generated by the program during the call, such as the content of the error log, into the span log.  Through the configuration file, you can control the log source (log4j2, logback, log4j), package name, and level. Please pay attention that  the original `log4j.xml`(or other log framework config files) has high priority than the `logconfig.properties`.

Reviewer 2: ```suggestion
`logger-plugin` can store the logs generated by the program during the call, such as the content of the error log, into the span log.  Through the configuration file, you can control the log source (log4j2, logback, log4j), package name, and level. Please note that the original `log4j.xml`(or other log framework config files) has a higher priority than the `logconfig.properties`.
```

# Configuration file

> By default, the configuration file is in `apache-skywalking-apm-bin/agent/config//logger-plugin/logconfig.properties`.Of course, **If the file does not exist, the configuration will still be configured with the following default values:**

Reviewer: ```suggestion
By default, the configuration file is in `apache-skywalking-apm-bin/agent/config//logger-plugin/logconfig.properties`.Of course, **If the file does not exist, the default configurations are as following,**
```

> By default, the configuration file is in `apache-skywalking-apm-bin/agent/config/logger-plugin/logconfig.properties`.Of course, **If the file does not exist, the default configurations are as following,**

Reviewer 2: ```suggestion
By default, the configuration file is in `apache-skywalking-apm-bin/agent/config/logger-plugin/logconfig.properties`. Of course, **If the file does not exist, the default configurations are as following:**
```

```properties
log4j.packages=*
log4j.level=error
log4j2.packages=*
log4j2.level=error
logback.packages=*
logback.level=error
```

The meaning of the above configuration is as follows:

1. SkyWalking opened the adaptor(bridge) between tracing kernel and log frameworks, including `log4j`, `log4j2`, `logback`.
2. Only collect logs at the `error` level, others would be ignored, including `trace`, `debug`, `info`, `warn`.
3. Wouldn't filter the logs by the package name.

# Property description

## packages

> **package attribute**: Specify the package name of the log that needs log conversion.  **The default value is `*` , and it will match all packages.**

Reviewer 2: ```suggestion
**package attribute**: Specify the package name of the log that needs log conversion. **The default value is `*`, and it will match all packages.**
```

packages value:

* the name of package, eg: `org.apache.skywalking`
* `*`:match all 

**Notice:**

> When matching multiple packages, the names of different packages should be separated by the comma.

Reviewer 2: ```suggestion
When matching multiple packages, the names of different packages should be separated by a comma.
```

## level

**level attribute** : The level of the log for conversion. **By default it is `error` level**.

The hierarchy order of log levels from low to high is as follows:

`trace` < `debug` < `info` <`warn`< `error` < `fatal`

**Notice:**

> Because `logback` not support `fatal`, it will be an exception if you set **logback.level=fatal**.

Reviewer: What would be the exception? Could you point me out? There is `string-switch` in your codes only

> Because `logback` not support `fatal`, it wouldn't work, and you need to look for error message in the log if you set **logback.level=fatal**.

Reviewer 2: ```suggestion
Because `logback` does not support the `fatal` log level, the highest level you may set it to is **logback.level=error**.
```
